src/IonTextReader.ts

```diff
@@ -89,13 +89,29 @@ export class TextReader implements Reader {
   }
 
     isIVM(input : string) : boolean {
-        if(this.depth() > 0) return false;
-        if(input.length < 6 || this.annotations().length > 0) return false;
+        if (this.depth() > 0) return false;
+        if (input.length < 6 || this.annotations().length > 0) return false;
         let prefix = "$ion_";
-        for(let i = 0; i < prefix.length; i++){
-            if(prefix.charAt(i) !== input.charAt(i)) return false;
+        let i = 0;
+
+        while (i < prefix.length) {
+            if (prefix.charAt(i) !== input.charAt(i)) return false;
+            i++;
+        }
+
+        while (i < input.length && input.charAt(i) != '_') {
```

Comment:
we could potentially combine this while loop and the next, returning false if a non-digit and non underscore character is encountered, but your approach ensures there's a single underscore, which is nice, too.  fine as-is